File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def make_binary(n, num):
    answer = [0 for _ in range(n)]
    idx = n - 1
    
    while num >= 1:
        answer[idx] = num % 2
        num //= 2
        idx -= 1
    
    return ''.join(map(str, answer))

print(make_binary(5, 9))
    
    
def solution(n, arr1, arr2):
    temp = [
        [0] * n
        for _ in range(n)
    ]
    
    answer = [
        "" for _ in range(n)
    ]
    
    for idx, elem in enumerate(arr1):
        binary_num = make_binary(n, elem)
        for i in range(n):
            temp[idx][i] = int(binary_num[i])
        
    for idx, elem in enumerate(arr2):
        binary_num = make_binary(n, elem)
        for i in range(n):
            if int(binary_num[i]) == 1:
                temp[idx][i] = int(binary_num[i])
            else:
                pass
    
    
    for i in range(n):
        for j in range(n):
            if temp[i][j] == 1:
                answer[i] += '#'
            else:
                answer[i] += ' '
        
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.js
function makeBinary(n, num) {
  const answer = Array(n).fill(0);
  let idx = n - 1;

  while (num >= 1) {
    answer[idx] = num % 2;
    num = Math.floor(num / 2);
    idx -= 1;
  }

  return answer.join("");
}

console.log(makeBinary(5, 9)); // "01001"


function solution(n, arr1, arr2) {
  const temp = Array.from({ length: n }, () => Array(n).fill(0));
  const answer = Array(n).fill("");

  arr1.forEach((elem, idx) => {
    const binaryNum = makeBinary(n, elem);
    for (let i = 0; i < n; i++) {
      temp[idx][i] = parseInt(binaryNum[i], 10);
    }
  });

  arr2.forEach((elem, idx) => {
    const binaryNum = makeBinary(n, elem);
    for (let i = 0; i < n; i++) {
      if (parseInt(binaryNum[i], 10) === 1) {
        temp[idx][i] = 1;
      }
    }
  });

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < n; j++) {
      answer[i] += temp[i][j] === 1 ? "#" : " ";
    }
  }

  return answer;
}